sml/src/sml_list.c

```diff
@@ -147,12 +156,57 @@ sml_list *sml_list_entry_parse(sml_buffer *buf) {
 	l->scaler = sml_i8_parse(buf);
 	if (sml_buf_has_errors(buf)) goto error;
 
+	value_tl = sml_buf_get_current_byte(buf);
```

Comment:
just a comment (was part of prev. code already): l doesn't get freed (sml_list_free(l)) if any of the prev. errors occurs.